paddle/py_paddle/dataprovider_converter.py

```diff
@@ -95,17 +98,27 @@ class DenseScanner(IScanner):
     def __init__(self, input_type, pos):
         IScanner.__init__(self, input_type, pos)
         self.__mat__ = None
+        self.__shape__ = None
         self.__height__ = 0
 
     def pre_scan(self, dat):
         self.__height__ += 1
 
-    def finish_pre_scan(self, argument):
+    def finish_pre_scan(self, argument, dat=None):
+        self.__shape__ = numpy.array(dat).shape
+        if len(self.__shape__) > 3:
+            raise ValueError("The dimension of input is greater than 3.")
+        dim = reduce(lambda x, y: x * y, self.__shape__)
+        if len(self.__shape__) == 1:
+            assert dim == self.input_type.dim
```

Comment:
```python
if len(self.__shape__) == 1 and dim != self.input_type.dim:
    raise ValueError(...)
```
Please use exception instead of assert. 

And it seems that this check should be in `pre_scan` method.